test/exec/test_on.cpp

```diff
@@ -0,0 +1,341 @@
+}
+TEST_CASE("exec::on can be called with const ref scheduler", "[adaptors][exec::on]") {
+  const inline_scheduler sched;
+  auto snd = exec::on(sched, ex::just(13));
+  auto op = ex::connect(std::move(snd), expect_value_receiver{13});
+  ex::start(op);
+  // The receiver checks if we receive the right value
+}
+TEST_CASE("exec::on can be called with ref scheduler", "[adaptors][exec::on]") {
+  inline_scheduler sched;
+  auto snd = exec::on(sched, ex::just(13));
+  auto op = ex::connect(std::move(snd), expect_value_receiver{13});
+  ex::start(op);
+  // The receiver checks if we receive the right value
+}
+
+TEST_CASE("exec::on forwards set_error calls", "[adaptors][exec::on]") {
+  error_scheduler<std::exception_ptr> sched{std::exception_ptr{}};
+  auto snd = exec::on(sched, ex::just(13));
+  auto op = ex::connect(std::move(snd), expect_error_receiver{});
+  ex::start(op);
+  // The receiver checks if we receive an error
+}
+TEST_CASE("exec::on forwards set_error calls of other types", "[adaptors][exec::on]") {
+  error_scheduler<std::string> sched{std::string{"error"}};
+  auto snd = exec::on(sched, ex::just(13));
+  auto op = ex::connect(std::move(snd), expect_error_receiver{std::string{"error"}});
+  ex::start(op);
+  // The receiver checks if we receive an error
+}
+TEST_CASE("exec::on forwards set_stopped calls", "[adaptors][exec::on]") {
+  stopped_scheduler sched{};
+  auto snd = exec::on(sched, ex::just(13));
+  auto op = ex::connect(std::move(snd), expect_stopped_receiver{});
+  ex::start(op);
+  // The receiver checks if we receive the stopped signal
+}
+
+TEST_CASE("exec::on has the values_type corresponding to the given values", "[adaptors][exec::on]") {
+  inline_scheduler sched{};
+
+  check_val_types<type_array<type_array<int>>>(exec::on(sched, ex::just(1)));
+  check_val_types<type_array<type_array<int, double>>>(exec::on(sched, ex::just(3, 0.14)));
+  check_val_types<type_array<type_array<int, double, std::string>>>(
+      exec::on(sched, ex::just(3, 0.14, std::string{"pi"})));
+}
+TEST_CASE("exec::on keeps error_types from scheduler's sender", "[adaptors][exec::on]") {
+  inline_scheduler sched1{};
+  error_scheduler sched2{};
+  error_scheduler<int> sched3{43};
+
+  check_err_types<type_array<std::exception_ptr>>(exec::on(sched1, ex::just(1)));
+  check_err_types<type_array<std::exception_ptr>>(exec::on(sched2, ex::just(2)));
+  check_err_types<type_array<std::exception_ptr, int>>(exec::on(sched3, ex::just(3)));
+}
+TEST_CASE("exec::on keeps sends_stopped from scheduler's sender", "[adaptors][exec::on]") {
+  inline_scheduler sched1{};
+  error_scheduler sched2{};
+  stopped_scheduler sched3{};
+
+  check_sends_stopped<false>(exec::on(sched1, ex::just(1)));
+  check_sends_stopped<true>(exec::on(sched2, ex::just(2)));
+  check_sends_stopped<true>(exec::on(sched3, ex::just(3)));
+}
+
+TEST_CASE("exec::on transitions back to the receiver's scheduler", "[adaptors][exec::on]") {
```

Comment:
Without a scheduler on the receiver, does `on(sched, closure)` complete on sched? The test above doesn't fail to compile in that situation, so I assume it does without any other alternative.

Reply:
Yes